tests/filtered_relation/tests.py

```diff
@@ -1044,6 +1045,18 @@ def test_condition_self_ref(self):
             [self.book1],
         )
 
+    def test_recursion_error(self):
+        qs = Book.objects.alias(
+            specific_mission=FilteredRelation(
+                "author__favorite_books", condition=Q(author__content_type__name="test")
+            )
+        )
+        self.assertIsInstance(qs, QuerySet)
+        try:
+            list(qs)
```

Comment:
You should assert against the returned values, from what I can see the results will be wrong as you assign an empty `WhereNode` on recursion errors.